pkg/operator/controller/ingress/monitoring.go

```diff
@@ -75,10 +100,13 @@ func desiredServiceMonitor(ic *operatorv1.IngressController, svc *corev1.Service
 		Version: "v1",
 	})
 	sm.SetOwnerReferences([]metav1.OwnerReference{deploymentRef})
-	return sm
+	return true, sm
```

Comment:
If we always want one, why return a boolean at all?

Reply:
Consistency.

Reply:
Consistency at what cost? Doesn't it mean that half of the new code in the reconcile logic is actually unreachable?

https://github.com/openshift/cluster-ingress-operator/pull/289/files#diff-b3b3514d6621705093f820e6f9feead0R34-R41

Reply:
Some of the code is unreachable.  For now, we are open-coding the same pattern for every resource, but we want eventually to refactor all the ensure methods with the common logic (the ensure method itself) abstracted away from the resource-specific logic (get current X, get desired X, create, update, delete X).  If this isn't useful, I can delete the delete logic and simplify the switch to 2 cases.

Reply:
Fair enough. I'll leave it up to you. Remove hold if you want to keep as is.

/hold

Reply:
/lgtm